### PR TITLE
create SpringBoot Custom Plugin Template

### PR DESCRIPTION
**Pull request by Luca-Ferrera**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/.idea/*
+/target/*
+*.iml
```
Comment (fredmaggiowski) [inline .gitignore:3]: Add DS_Store, build and intermediates directories etc.

File: config/Dockerimage [hunk 2]
```diff
@@ -0,0 +1 @@
+%NEXUS_HOSTNAME%/%CUSTOM_PLUGIN_IMAGE_NAME%
```
Comment (fredmaggiowski) [inline config/Dockerimage]: This file is not needed anymore, @JGiola am I right?

File: src/main/java/eu/miaplatform/customplugin/springboot/lib/CPController.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+
+    @GetMapping("/-/ready")
+    @ApiOperation(value = "Ready")
+    public ResponseEntity ready(@ApiIgnore @ModelAttribute(CP_REQUEST) CPRequest cpRequest) {
+        return readinessHandler(cpRequest);
+    }
+
+
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/lib/CPController.java]:
```suggestion
```

File: src/main/java/eu/miaplatform/customplugin/springboot/Application.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class Application {
+
+    //Default server port is 3000
+    //Run mvn spring-boot:run -Dspring-boot.run.arguments=--server.port=8000 to change server port
+    public static void main(String[] args) {
+
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/Application.java]:
```suggestion
```

File: src/main/java/eu/miaplatform/customplugin/springboot/SwaggerConfig.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    public String getContactUrl() {
+        return "https://www.mia-platform.eu";
+    }
+
+    @Override
+    public String getContactEmail() {
+        return "info@mia-platform.eu";
+    }
+
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/SwaggerConfig.java:66]:
```suggestion
```

File: src/main/java/eu/miaplatform/customplugin/springboot/SwaggerConfig.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/SwaggerConfig.java]:
```suggestion
```

File: src/main/java/eu/miaplatform/customplugin/springboot/lib/CPDecorator.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+            filterChain.doFilter(requestWrapper, responseWrapper);
+
+            postHandler.accept(responseWrapper);
+        });
+        registration.setName(filter + number);
+        number++;
+        registration.addUrlPatterns(path);
+        return registration;
+    }
+
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/lib/CPDecorator.java]:
```suggestion
```

File: src/main/java/eu/miaplatform/customplugin/springboot/lib/CPRequestWrapper.java [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+    @Override
+    public Enumeration<String> getParameterNames() {
+        return Collections.enumeration(getParameterMap().keySet());
+    }
+
+    @Override
+    public String[] getParameterValues(final String name) {
+        return getParameterMap().get(name);
+    }
+
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/lib/CPRequestWrapper.java]:
```suggestion
```

File: src/main/java/eu/miaplatform/customplugin/springboot/lib/FilterServletInputStream.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+package eu.miaplatform.customplugin.springboot.lib;
+
+import javax.servlet.ReadListener;
+import javax.servlet.ServletInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class FilterServletInputStream extends ServletInputStream {
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/lib/FilterServletInputStream.java]: What's the purpose of this class?

File: src/main/java/eu/miaplatform/customplugin/springboot/lib/FilterServletOutputStream.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public boolean isReady() {
+        return false;
+    }
+
+    @Override
+    public void setWriteListener(WriteListener writeListener) {
+
+    }
+}
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/lib/FilterServletOutputStream.java]:
```suggestion
}

```

File: src/main/java/eu/miaplatform/customplugin/springboot/lib/LoggingInterceptor.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+@Component
+public class LoggingInterceptor implements HandlerInterceptor {
+
+    private final Logger logger = LoggerFactory.getLogger(LoggingInterceptor.class);
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+
+        //logger.info("preHandle - request [" + request + "]");
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/lib/LoggingInterceptor.java]: Remove commented code.

File: src/main/java/eu/miaplatform/customplugin/springboot/lib/LoggingInterceptor.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+        while (headerNames.hasMoreElements()) {
+            logger.info("headerName = " + headerNames.nextElement());
+        }*/
+        return true;
+    }
+
+    @Override
+    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler, ModelAndView modelAndView) throws Exception {
+
+        //logger.info("postHandle - response [" + response + "]");
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/lib/LoggingInterceptor.java]: Remove commented code

File: src/main/java/eu/miaplatform/customplugin/springboot/lib/Options.java [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    private String backofficeHeaderKey;
+    private String microserviceGatewayServiceName;
+
+    public Options() {
+        this.userIdHeaderKey = "USERID_HEADER_KEY";
+        this.groupsHeaderKey = "GROUPS_HEADER_KEY";
+        this.clientTypeHeaderKey = "CLIENTTYPE_HEADER_KEY";
+        this.backofficeHeaderKey = "BACKOFFICE_HEADER_KEY";
+        this.microserviceGatewayServiceName = "MICROSERVICE_HEADER_KEY";
+    }
```
Comment (fredmaggiowski) [inline src/main/java/eu/miaplatform/customplugin/springboot/lib/Options.java]:
```suggestion
    }

```

File: tag.sh [hunk 14]
```diff
@@ -0,0 +1,87 @@
+# * Copyright 2019 Mia srl
```
Comment (fredmaggiowski) [inline tag.sh]: Remove *

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]

**Review by Antolas — APPROVED** [review]

**Pull request merged**